src/main/java/dinkplugin/notifiers/CombatTaskNotifier.java

```diff
@@ -41,19 +77,49 @@ private void handle(CombatAchievementTier tier, String task) {
         if (tier.ordinal() < config.minCombatAchievementTier().ordinal())
             return;
 
-        String player = Utils.getPlayerName(client);
-        String message = StringUtils.replaceEach(
-            config.combatTaskMessage(),
-            new String[] { "%USERNAME%", "%TIER%", "%TASK%", "%POINTS%" },
-            new String[] { player, tier.toString(), task, String.valueOf(tier.getPoints()) }
-        );
-
-        createMessage(config.combatTaskSendImage(), NotificationBody.<CombatAchievementData>builder()
-            .type(NotificationType.COMBAT_ACHIEVEMENT)
-            .text(message)
-            .playerName(player)
-            .extra(new CombatAchievementData(tier, task, tier.getPoints()))
-            .build());
+        // delay notification for varbits to be updated
+        clientThread.invokeAtTickEnd(() -> {
+            int taskPoints = tier.getPoints();
+            int totalPoints = client.getVarbitValue(TOTAL_POINTS_ID);
+
+            Integer nextUnlockPointsThreshold = cumulativeUnlockPoints.ceilingKey(totalPoints + 1);
+            Map.Entry<Integer, CombatAchievementTier> prev = cumulativeUnlockPoints.floorEntry(totalPoints);
+            int prevThreshold = prev != null ? prev.getKey() : 0;
+
+            Integer tierProgress, tierTotalPoints;
+            if (nextUnlockPointsThreshold != null) {
+                tierProgress = totalPoints - prevThreshold;
+                tierTotalPoints = nextUnlockPointsThreshold - prevThreshold;
+            } else {
+                tierProgress = tierTotalPoints = null;
+            }
+
+            boolean crossedThreshold = prevThreshold > 0 && totalPoints - taskPoints < prevThreshold;
+            CombatAchievementTier completedTier = crossedThreshold ? prev.getValue() : null;
+            String completedTierName = completedTier != null ? completedTier.getDisplayName() : "N/A";
+
+            String player = Utils.getPlayerName(client);
+            String message = StringUtils.replaceEach(
+                crossedThreshold ? config.combatTaskUnlockMessage() : config.combatTaskMessage(),
+                new String[] { "%USERNAME%", "%TIER%", "%TASK%", "%POINTS%", "%TOTAL_POINTS%", "%COMPLETED%" },
+                new String[] { player, tier.toString(), task, String.valueOf(taskPoints), String.valueOf(totalPoints), completedTierName }
```

Comment:
I see we always include a replacement for `%COMPLETED%` - I am perfectly fine with this solution :+1: